lib/ZnapZend/ZFS.pm

```diff
@@ -54,7 +54,12 @@ my $splitHostDataSet = sub {
 };
 
 my $splitDataSetSnapshot = sub {
-    return ($_[0] =~ /^([^\@]+)\@([^\@]+)$/);
+    my $count = ($_[0] =~ tr/@//);
```

Comment:
This, I think, will actually modify the $_[0] with this the argument given when calling the function, this is unwanted action at a distance I think. I would suggest to work with a copy

Reply:
`my $snap  = shift;`